### ISSUE TITLE
Main branch contains older ECR helper than 1.7.0

### ISSUE THREAD

**Issue by Vrtak-CZ** [GoogleContainerTools/kaniko#1879]
**Actual behavior**
I do not understand the reason why ECR Credentials Helper got downgraded from 0.5.0 to 0.4.0. The 0.4.0 does not support ECR public registry (0.5.0 does).

```
ella:~:% podman run --rm --entrypoint "" gcr.io/kaniko-project/executor:v1.7.0-debug /kaniko/docker-credential-ecr-login -v    

amazon-ecr-credential-helper
Version:    0.5.0
Git commit: b94b92f
```

The change was introduced here https://github.com/GoogleContainerTools/kaniko/commit/f6942123850aef0a614cfda5bdb2643363852488

**Expected behavior**
Upgrade main branch back to 0.5.0 to support ECR Public registry

**To Reproduce**
1. https://github.com/GoogleContainerTools/kaniko/blob/986121553be04f5cd23f96e0d0443cd6212cfb25/deploy/Dockerfile#L29
2. https://github.com/GoogleContainerTools/kaniko/blob/986121553be04f5cd23f96e0d0443cd6212cfb25/deploy/Dockerfile_debug#L29
 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ ] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [ ] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [ ] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ ] </li></ul>|

**Comment by imjasonh** [comment]
Weird, I'm not sure why I did that either. Fixed in #1880, and I agree we should do it before cutting a new release.

Thanks for noticing this! 👍

**Comment by imjasonh** [comment]
FWIW #1872 should also help, since our dependency on cred helpers will be tracked in go.mod, and Dependabot can propose upgrades for us.
